src/cfg/liveness-traversal.h

```diff
@@ -171,30 +172,18 @@ struct LivenessWalker : public CFGWalker<SubType, VisitorType, Liveness> {
     return nullptr;
   }
 
-  // If there are too many locals, we cannot run currently as
-  // numLocals * numLocals might overflow. We may want to add an option for
-  // a sparse matrix at some point TODO
-  bool canRun(Function* func) {
-    Index numLocals = func->getNumLocals();
-    if (uint64_t(numLocals) * uint64_t(numLocals) <=
-        std::numeric_limits<Index>::max()) {
-      return true;
-    }
-    std::cerr << "warning: too many locals (" << numLocals
-              << ") to run liveness analysis in " << this->getFunction()->name
-              << '\n';
-    return false;
-  }
+  bool canRun(Function*) { return true; }
 
   // main entry point
 
   void doWalkFunction(Function* func) {
     numLocals = func->getNumLocals();
-    assert(canRun(func));
-    copies.resize(numLocals * numLocals);
-    std::fill(copies.begin(), copies.end(), 0);
+    copies.recreate(numLocals);
+    std::fill(totalCopies.begin(),
+              totalCopies.begin() +
+                std::min<uint64_t>(totalCopies.size(), numLocals),
+              0);
     totalCopies.resize(numLocals);
```

Comment:
The code from before looks simpler I think, where it resizes first, then does a fill on the entire thing after that - was there a reason not to do it that way?

Reply:
It is redundant work. After resizing, the resized elements will be zeros already, so one only needs to clear the old entries.

Reply:
Adjusted to do a vector.clear() followed by a vector.resize(), maybe that looks the simplest. And looks like is quite close to the same performance. Authored a second PR #4580 to systematically do the same to other places as well.